app/src/main/java/com/pokemonshowdown/data/ItemDex.java

```diff
@@ -71,7 +89,7 @@ public static int getItemIcon(Context appContext, String itemName) {
                 .getIdentifier("item_" + MyApplication.toId(itemName), "drawable", appContext.getPackageName());
     }
 
-    public HashMap<String, String> getItemDexEntries() {
+    public HashMap<String, Integer> getItemDexEntries() {
```

Comment:
This is only needed for SearchableActivity. So it's pretty simple to lazy load information. Instead of getItemDexEntries, the `ItemAdapter` only needs to know how many elements there are. During `getView()` in ItemAdapter, we can get key from the `string-array` and correlated values, and then populate view as the users scroll down. Similarly for `searchItem()`, we can just load from resource. Best part is that this won't trigger garbage collector nor taking up our memory.

This is pretty trivial just for ItemDex but it'll get worse when we are storing PokeDex in memory and that's why Im changing it up